### PR TITLE
Update demo script

### PR DESCRIPTION
**Pull request by ableuler**
Make it possible to run the demo script against any renku instance, namely staging or a locally running minikube deployment:
- Allow ssh port other than 22 for git remote
- Clean up (reduce) dependencies
- Update readme with examples on how to run demo against any renku instance

(Integration test on renku is still passing after manually pushing the content of this PR to renku/renku-demo:latest)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,3 +21,18 @@ and adapted in `users.json`.
 from your gitlab instance.
 If you want to build your image locally (for example because you want to change the pre-configured
 users in users.json), run `docker build . -t renku/renku-demo`
+
+### Running demo against any renku instance
+Exmaples:
```
Comment (rokroskar) [inline README.md]: typo

File: README.md [hunk 2]
```diff
@@ -21,3 +21,18 @@ and adapted in `users.json`.
 from your gitlab instance.
 If you want to build your image locally (for example because you want to change the pre-configured
 users in users.json), run `docker build . -t renku/renku-demo`
+
+### Running demo against any renku instance
```
Comment (rokroskar) [inline README.md]: "Running **the** demo"

File: README.md [hunk 3]
```diff
@@ -21,3 +21,18 @@ and adapted in `users.json`.
...
+Exmaples:
+- `docker run -it --rm --network host \`<br> 
+   `-e GITLAB_URL=http://$(minikube ip)/gitlab \`<br> 
+   `-e KEYCLOAK_URL=http://$(minikube ip)  \`<br> 
+   `renku/renku-demo:latest`
+
+- `docker run -it --rm --network host \`<br> 
+   `-e GITLAB_URL=https://staging.testing.datascience.ch/gitlab \`<br>
```
Comment (rokroskar) [inline README.md]: maybe don't show the staging URL in public?

File: steps/4-initialize-repo.sh [hunk 4]
```diff
@@ -47,8 +47,8 @@ git add .
 git commit -m "Added readme"
 
 if ! [ -z "$DOCKER" ]; then
-    export GITLAB_PORT=`echo $REMOTE_REPO_URL | grep  -o '\:[0-9]*' | grep -o '[0-9]\+'`
-    export GITLAB_HOST=`echo $GITLAB_URL | sed -e 's/http\(s\)\{0,1\}\:\/\///g'`
+    export GITLAB_PORT=$(python ../utils/get_ssh_port.py)
```
Comment (rokroskar) [inline steps/4-initialize-repo.sh]: could these two be combined into one script that takes a `--port` and `--host-name` flags using `argparse` or `click`?

File: utils/get_host_name.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re, os
+
+print(re.search(r'https?://([^\/]*)', os.environ['GITLAB_URL']).group(1))
```
Comment (rokroskar) [inline utils/get_host_name.py]: this will work only with `https`?

Comment (ableuler) [reply utils/get_host_name.py]: No, the question mark denotes 0 or one occurrences of the previous character.

File: steps/4-initialize-repo.sh (hunk 4 above)
Comment (ableuler) [reply steps/4-initialize-repo.sh]: okok, I will make it a little nicer...

File: README.md (hunk 3 above)
Comment (ableuler) [reply README.md]: Ok, didn't see this as a problem but I will remove it.

File: utils/get_host_name.py (hunk 5 above)
Comment (rokroskar) [reply utils/get_host_name.py]: oops, missed that detail :)

**Pull request merged**